lib/si-frontend-types-rs/src/newhotness/component_connections.rs

```diff
@@ -0,0 +1,84 @@
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use si_events::workspace_snapshot::EntityKind;
+use si_frontend_types_macros::{
+    FrontendChecksum,
+    FrontendObject,
+    MV,
+    Refer,
+};
+use si_id::{
+    AttributeValueId,
+    ChangeSetId,
+    ComponentId,
+    InputSocketId,
+    OutputSocketId,
+    PropId,
+};
+
+use crate::reference::{
+    Reference,
+    ReferenceKind,
+};
+
+#[derive(
+    Clone, Debug, Deserialize, Eq, PartialEq, Ord, PartialOrd, Serialize, FrontendChecksum,
+)]
+#[remain::sorted]
+#[serde(rename_all = "camelCase", tag = "kind")]
+pub enum Connection {
+    Prop {
+        from_component_id: ComponentId,
```

Comment:
When we were talking about the idea of "weak references" being a thing, I was talking about it just like `Reference`, only without the `checksum`. My expectation was:
```
from_component_id: WeakReference<Component>
```
I want to explicitly call out that this ID correlated to the `EntityId` of a different `EntityKind`. I want the entity kind _in_ the `WeakReference` data.

We can ship this for now.. but when we drop "Beta" from this. Let's also create a `WeakReference`. That way, in `webworker.ts` we can actually write re-usable code that checks _every_ MV for `WeakReference` and knows which `EntityKind` to look up. That way we don't have to keep writing `if` blocks every time.

Reply:
100%. That will be a fast follow. Good call out.